tests/main/test_retriever_agent.py

```diff
@@ -0,0 +1,205 @@
+import random
+import warnings
+from typing import Dict, List, Union
+
+import pytest
+
+from langroid.agent.special.retriever_agent import (
+    RecordDoc,
+    RecordMetadata,
+    RetrieverAgent,
+    RetrieverAgentConfig,
+)
+from langroid.agent.task import Task
+from langroid.cachedb.redis_cachedb import RedisCacheConfig
+from langroid.embedding_models.models import OpenAIEmbeddingsConfig
+from langroid.language_models.openai_gpt import OpenAIChatModel, OpenAIGPTConfig
+from langroid.mytypes import Entity
+from langroid.parsing.parser import ParsingConfig, Splitter
+from langroid.prompts.prompts_config import PromptsConfig
+from langroid.utils.configuration import Settings, set_global
+from langroid.utils.system import rmdir
+from langroid.vector_store.qdrantdb import QdrantDBConfig
+
+storage_path = ".qdrant/testdata1"
+rmdir(storage_path)
+
+
+def generate_data(size: int) -> List[Dict[str, Union[int, float, str]]]:
+    # Create a list of states
+    states = ["CA", "TX"]
+
+    # Generate random age between 18 and 100
+    ages = [random.randint(18, 100) for _ in range(size)]
+
+    # Generate random gender
+    genders = [random.choice(["Male", "Female"]) for _ in range(size)]
+
+    # Generate random state
+    states_col = [random.choice(states) for _ in range(size)]
+
+    # Generate random height between 4.0 and 6.5
+    heights = [round(random.uniform(4.0, 6.5), 2) for _ in range(size)]
+
+    random_people = [
+        {
+            "id": i + 1,
+            "age": ages[i],
+            "gender": genders[i],
+            "state": states_col[i],
+            "height": heights[i],
+        }
+        for i in range(size)
+    ]
+
+    return random_people
+
+
+class PeopleMetadata(RecordMetadata):
+    source: str = None
+    id: int
+
+
+class PeoplyRecord(RecordDoc):
+    metadata: PeopleMetadata
+
+
+set_global(
+    Settings(
+        cache=True,
+        gpt3_5=True,
+    )
+)  # allow cacheing
+
+known_people = [
+    {"id": 102, "age": 99, "gender": "male", "state": "NY", "height": "6.1ft"},
+    {
+        "id": 103,
+        "age": 98,
+        "gender": "female",
+        "state": "NY",
+        "height": "5.7ft",
+    },
+    {
+        "id": 104,
+        "age": 97,
+        "gender": "female",
+        "state": "NY",
+        "height": "5.8ft",
+    },
+]
+
+people = generate_data(100) + known_people
+
+
+class _TestRetrieverAgent(RetrieverAgent):
+    def __init__(self, config: RetrieverAgentConfig):
+        super().__init__(config)
+        self.config = config
+
+    def get_records(self) -> List[PeoplyRecord]:
+        people_rec = []
+        row_num = 0
+        for person in people:
+            full_row = f"""
+            {person['age']}|{person['gender']}|{person['state']}|{person['height']}
```

Comment:
If you are not using the words `age`, `gender`, `state` etc in the record, how do you expect the near-neighbor retrieval to work?  We should store  something like "age = ... | gender = ..."

Reply:
I see, had this thought, but was mimicking the `segmentor` example. 

<img width="833" alt="image" src="https://github.com/langroid/langroid/assets/15859139/4db4195a-cf80-491e-9770-e4f53075fb66">